### PR TITLE
Add a chat with us button to the purchase checkout page

### PR DESCRIPTION
**Pull request by omarjackman**
This pull request adds a "Chat with us" button to the checkout screen so that we can offer users the chance to chat with a Happiness Engineer before making their purchase.

For now the button is only offered on the variation of the checkout page used for paying with credits and it sits behind the `upgrades/presale-chat` feature flag.

### How to test
1. Add credits enough credits to your account to make a purchase.
2. Navigate to http://calypso.localhost:3000/plans
3. From the browser console enter: `localStorage.setItem( 'ABTests', '{"presaleChatButton_20161129":"showChatButton"}' )`. This will allow us to bypass the abtest
4. Select an upgrade
5. Notice on the checkout screen you will see a new button on the bottom right labeled "Need help? Chat with us"
6. Click the chat with us button.
7. Within tracks notice that a ping has been made for `calypso_chat_button_click`
8. Start a chat.
9. Within olark notice that the user is tagged with "Context: presale"
10. Check tracks and notice that another ping has made for `calypso_chat_button_chat_begin`

### What to expect
screen shot 2016-11-22 at 4 30 43 pm

### CODE REVIEW

**Comment by omarjackman** [comment]
cc @dllh @evilluendas

**Comment by evilluendas** [comment]
It looks and works as expected, nice work!

**Comment by dllh** [comment]
Hmm, just tested and it's not working for me. Are credits required in order for it to work? I guess it's also possible no operators were available. I also note though that when you try to use a credit card, there's a Paypal option out to the right where the chat prompt lives in your screen shot. Will they conflict?

screen shot 2016-11-23 at 9 41 12 am

**Comment by dllh** [comment]
It works when I have credits (only paid, vs. free, though; it's a separate issue unrelated to this work, but I wonder if that's expected?), but that'll be the exception rather than the rule, so we'll need to test that and probably figure out the Paypal button conflict before we can proceed.

**Comment by omarjackman** [comment]
> Are credits required in order for it to work?

Yes

> I also note though that when you try to use a credit card, there's a Paypal option out to the right where the chat prompt lives in your screen shot. Will they conflict?

Yes :)

Because I have a bunch of credits the first variation of the page I saw was the "pay using credits" so I started with that. We might need to tweak the placement or consider a redesign in the credit card/paypal case. @evilluendas what do you think we could do here?

**Comment by evilluendas** [comment]
OH! I never saw the PayPal link.

I think it makes sense to move the PayPal link next to the Pay button, so we have "Pay $63 or use PayPal" and then the "Chat with us" button on the right. 

We could move the Chat button to the left instead, but then it wouldn't make sense ("Pay", "chat", and then "or use PayPal").

**Comment by evilluendas** [comment]
A quick mockup of what I meant before:

pre-purchase-paypal

**Comment by breezyskies** [comment]
This looks great @evilluendas! I mentioned in Slack to @omarjackman, but noting here as well -- this change is probably worth a/b testing for a few reasons:

- To get a gauge the resulting support volume before opening to 100% of users in this flow.
- To measure the impact the chat button presence / usage has on checkout conversions. 
- Just generally speaking, it's a checkout flow change, so always good to keep an eye on the impact any changes we make have.

**Comment by omarjackman** [comment]
I just realized while trying to add the abtest that this PR doesn't affect that screen. I'll add another PR for the credit card purchase screen separately and add the AB test there.

**Comment by omarjackman** [comment]
@dllh I added an abtest for this. Do you think this can ship?

File: client/lib/abtest/active-tests.js [hunk 1]
```diff
@@ -125,5 +125,14 @@ module.exports = {
 		defaultVariation: 'showPlansAfterAuth',
 		allowExistingUsers: true
 	},
-};
 
+	presaleChatButton: {
+		datestamp: '20161129',
+		variations: {
+			showChatButton: 80,
+			original: 20
```
Comment (dllh) [inline client/lib/abtest/active-tests.js]: Maybe rename here and below to `noChatButton` or something more descriptive than `original`? I'm not sure that's essential, and if using something like `original` is how things are done elsewhere, that's fine. It just occurred to me that reading through the code and imagining trying to look at data for this, `original` isn't super meaningful over time.

Comment (omarjackman) [inline client/lib/abtest/active-tests.js]: @dllh I originally had `hideChatButton` but saw the usage of [`original`](https://github.com/Automattic/wp-calypso/blob/master/client/lib/abtest/active-tests.js#L85) in a few places and decided to go with the flow.

**Pull request closed without merging**